pkg/pipelines/environment.go

```diff
@@ -0,0 +1,84 @@
+package pipelines
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/openshift/odo/pkg/pipelines/meta"
+	"github.com/openshift/odo/pkg/pipelines/roles"
+)
+
+const (
+	envNamespace   = "namespace.yaml"
+	envRoleBinding = "rolebinding.yaml"
+)
+
+// EnvParameters encapsulates parameters for add env command
+type EnvParameters struct {
+	EnvName    string
+	Output     string
+	Prefix     string
+	GitOpsRepo string
+}
+
+// Env will bootstrap a new environment directory
+func Env(o *EnvParameters) error {
+
+	gitopsName := getGitopsRepoName(o.GitOpsRepo)
+	gitopsPath := filepath.Join(o.Output, gitopsName)
+
+	envName := addPrefix(o.Prefix, o.EnvName)
+	envPath := getEnvPath(gitopsPath, o.EnvName, o.Prefix)
+
+	// check if the gitops dir exists
+	exists, _ := isExisting(gitopsPath)
+	if !exists {
+		return fmt.Errorf("%s doesn't exist at %s", gitopsName, o.Output)
+	}
+
+	// check if the environment dir already exists
+	exists, _ = isExisting(envPath)
+	if exists {
+		return fmt.Errorf("%s already exists", envName)
+	}
+
+	err := addKustomize("resources", []string{envNamespace, envRoleBinding}, filepath.Join(envPath, "base", kustomize))
+	if err != nil {
+		return err
+	}
+
+	err = addKustomize("bases", []string{"../base"}, filepath.Join(envPath, "overlays", kustomize))
+	if err != nil {
+		return err
+	}
+
+	if err = addEnvResources(o.Prefix, envPath, envName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addEnvResources(prefix, envPath, envName string) error {
+
+	namespaces := namespaceNames(prefix)
+
+	outputs := map[string]interface{}{}
+	basePath := filepath.Join(envPath, "base")
+
+	outputs[envNamespace] = createNamespace(envName)
+
+	sa := roles.CreateServiceAccount(meta.NamespacedName(namespaces["cicd"], saName))
+
+	outputs[envRoleBinding] = roles.CreateRoleBinding(meta.NamespacedName(envName, roleBindingName), sa, "ClusterRole", roles.ClusterRoleName)
+
```

Comment:
SA (pipeline) in cicd-environment needs ClusterRole Edit in the target env namespace in order to deploy.   See https://github.com/rhd-gitops-example/odo/blob/43f596c8f00b12f7accbc28e9a8aa7ce24f383c1/pkg/pipelines/bootstrap.go#L140

Reply:
made the changes